Option to disable "registry.quarkus.io"

### Describe the bug

My company has a strict proxy policy, so we got log of errors with the DEV UI and Quarkus Ext. Registry.

Is it possible to disable it ?


```
Caused by: io.quarkus.registry.RegistryResolutionException: Failed to resolve the Quarkus extension registry descriptor of registry.quarkus.io from registry.quarkus.io (https://registry.quarkus.io/maven)
	at io.quarkus.registry.client.maven.MavenRegistryClientFactory.buildRegistryClient(MavenRegistryClientFactory.java:101)
	at io.quarkus.registry.ExtensionCatalogResolver$Builder.buildRegistryClients(ExtensionCatalogResolver.java:120)
	at io.quarkus.registry.ExtensionCatalogResolver$Builder.build(ExtensionCatalogResolver.java:89)
	at io.quarkus.devtools.project.QuarkusProjectHelper.getCatalogResolver(QuarkusProjectHelper.java:195)
	at io.quarkus.devtools.project.buildfile.MavenProjectBuildFile.getProject(MavenProjectBuildFile.java:75)
	at io.quarkus.devtools.project.buildfile.MavenProjectBuildFile.getProject(MavenProjectBuildFile.java:66)
	at io.quarkus.devtools.project.QuarkusProjectHelper.getCachedProject(QuarkusProjectHelper.java:59)

```

### Expected behavior

An option to disable Quarkus Registry Extension in Dev UI

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java
@@ -53,6 +53,12 @@ public interface DevUIConfig {
     Cors cors();
 
     /**
+     * Enable/Disable the ability to add and remove extensions from Dev UI
+     */
+    @WithDefault("true")
+    boolean allowExtensionManagement();
+
+    /**
      * Fine tune the theme
      */
     Optional<Theme> theme();

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/ExtensionsProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/ExtensionsProcessor.java
@@ -21,6 +21,7 @@ import io.quarkus.devtools.commands.data.QuarkusCommandException;
 import io.quarkus.devtools.commands.data.QuarkusCommandOutcome;
 import io.quarkus.devtools.project.QuarkusProject;
 import io.quarkus.devtools.project.QuarkusProjectHelper;
+import io.quarkus.devui.deployment.DevUIConfig;
 import io.quarkus.devui.deployment.ExtensionsBuildItem;
 import io.quarkus.devui.deployment.InternalPageBuildItem;
 import io.quarkus.devui.deployment.extension.Extension;
@@ -34,7 +35,7 @@ import io.quarkus.devui.spi.page.Page;
 public class ExtensionsProcessor {
 
     @BuildStep(onlyIf = IsLocalDevelopment.class)
-    InternalPageBuildItem createExtensionsPages(ExtensionsBuildItem extensionsBuildItem) {
+    InternalPageBuildItem createExtensionsPages(ExtensionsBuildItem extensionsBuildItem, DevUIConfig config) {
 
         InternalPageBuildItem extensionsPages = new InternalPageBuildItem("Extensions", 10, "qwc-extensions-menu-action");
 
@@ -52,15 +53,18 @@ public class ExtensionsProcessor {
                 .icon("font-awesome-solid:puzzle-piece")
                 .componentLink("qwc-extensions.js"));
 
+        extensionsPages.addBuildTimeData("allowExtensionManagement", config.allowExtensionManagement());
+
         return extensionsPages;
     }
 
     @BuildStep(onlyIf = IsLocalDevelopment.class)
     void createBuildTimeActions(BuildProducer<BuildTimeActionBuildItem> buildTimeActionProducer,
-            LaunchModeBuildItem launchModeBuildItem) {
+            LaunchModeBuildItem launchModeBuildItem, DevUIConfig config) {
 
         if (launchModeBuildItem.getDevModeType().isPresent()
-                && launchModeBuildItem.getDevModeType().get().equals(DevModeType.LOCAL)) {
+                && launchModeBuildItem.getDevModeType().get().equals(DevModeType.LOCAL)
+                && config.allowExtensionManagement()) {
 
             BuildTimeActionBuildItem buildTimeActions = new BuildTimeActionBuildItem(NAMESPACE);
             getCategories(buildTimeActions);

--- a/extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qwc/qwc-extensions.js
+++ b/extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qwc/qwc-extensions.js
@@ -3,6 +3,7 @@ import { unsafeHTML } from 'lit/directives/unsafe-html.js';
 import { RouterController } from 'router-controller';
 import { devuiState } from 'devui-state';
 import { observeState } from 'lit-element-state';
+import { allowExtensionManagement } from 'devui-data';
 import 'qwc/qwc-extension.js';
 import 'qwc/qwc-extension-link.js';
 import 'qwc/qwc-extension-add.js';
@@ -96,14 +97,16 @@ export class QwcExtensions extends observeState(LitElement) {
     connectedCallback() {
         super.connectedCallback();
         window.addEventListener('extensions-filters-changed', this._onFiltersChanged);
-        this.jsonRpc.getInstalledNamespaces().then(jsonRpcResponse => {
-            if (jsonRpcResponse.result) {
-                this._installedExtensions = jsonRpcResponse.result;
-                this._addExtensionsEnabled = true;
-            }
-        }).catch(e => {
-            notifier.showErrorMessage("Could not list namespaces "+ e?.error?.message);
-        });
+        if (allowExtensionManagement) {
+            this.jsonRpc.getInstalledNamespaces().then(jsonRpcResponse => {
+                if (jsonRpcResponse.result) {
+                    this._installedExtensions = jsonRpcResponse.result;
+                    this._addExtensionsEnabled = true;
+                }
+            }).catch(e => {
+                notifier.showErrorMessage("Could not list namespaces "+ e?.error?.message);
+            });
+        }
     }
 
     disconnectedCallback() {

